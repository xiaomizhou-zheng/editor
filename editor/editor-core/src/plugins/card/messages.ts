import { defineMessages } from 'react-intl-next';

export const messages = defineMessages({
  url: {
    id: 'fabric.editor.url',
    defaultMessage: 'Display URL',
    description: 'Convert the card to become a regular text-based hyperlink.',
  },
  block: {
    id: 'fabric.editor.displayBlock',
    defaultMessage: 'Display card',
    description:
      'Display link as a card with a rich preview similar to in a Facebook feed with page title, description, and potentially an image.',
  },
  inline: {
    id: 'fabric.editor.displayInline',
    defaultMessage: 'Display inline',
    description: 'Display link with the title only.',
  },
  embed: {
    id: 'fabric.editor.displayEmbed',
    defaultMessage: 'Display embed',
    description: 'Display link as an embedded object',
  },
  link: {
    id: 'fabric.editor.displayLink',
    defaultMessage: 'Display as text',
    description: 'Convert the card to become a regular text-based hyperlink.',
  },
  card: {
    id: 'fabric.editor.cardFloatingControls',
    defaultMessage: 'Card options',
    description: 'Options to change card type',
  },
  blockCardUnavailable: {
    id: 'fabric.editor.blockCardUnavailable',
    defaultMessage:
      'The inline link is inside {node} and cannot have its view changed',
    description:
      'Warning message to show the user that this node cannot change its view',
  },
  displayOptionUnavailableInParentNode: {
    id: 'fabric.editor.displayOptionUnavailableInParentNode',
    defaultMessage: "This display option isn't available inside {node}",
    description:
      'Warning message to show the user that this node option is not available inside a parent node type',
  },
  urlTitle: {
    id: 'fabric.editor.urlTitle',
    defaultMessage: 'URL',
    description:
      'Title for option to convert the card to become a regular text-based hyperlink.',
  },
  blockTitle: {
    id: 'fabric.editor.blockTitle',
    defaultMessage: 'Card',
    description: 'Title for option to display link in the card view.',
  },
  inlineTitle: {
    id: 'fabric.editor.inlineTitle',
    defaultMessage: 'Inline',
    description: 'Title for option to display link in the inline view.',
  },
  embedTitle: {
    id: 'fabric.editor.embedTitle',
    defaultMessage: 'Embed',
    description: 'Title for option to display link as an embedded object.',
  },
  urlDescription: {
    id: 'fabric.editor.urlDescription',
    defaultMessage: 'Display link as URL',
    description:
      'Description for option to convert the card to become a regular text-based hyperlink.',
  },
  blockDescription: {
    id: 'fabric.editor.blockDescription',
    defaultMessage:
      'Display more information about a link, including a summary and actions',
    description: 'Description for option to display link in the card view.',
  },
  inlineDescription: {
    id: 'fabric.editor.inlineDescription',
    defaultMessage: 'Display link as inline text',
    description: 'Description for option to display link in the inline view.',
  },
  embedDescription: {
    id: 'fabric.editor.ecombedDescription',
    defaultMessage: 'Display an interactive preview of a link',
    description:
      'Description for option to display link as an embedded object.',
  },
});
