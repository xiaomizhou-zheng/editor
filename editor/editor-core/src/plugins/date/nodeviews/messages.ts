import { defineMessages } from 'react-intl-next';

export const messages = defineMessages({
  editText: {
    id: 'fabric.editor.editDateText',
    defaultMessage: 'Edit Date',
    description: 'Title for the input that changes the date',
  },
});
