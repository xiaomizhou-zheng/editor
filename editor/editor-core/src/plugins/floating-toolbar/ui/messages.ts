import { defineMessages } from 'react-intl-next';

export default defineMessages({
  confirmModalDefaultHeading: {
    id: 'fabric.editor.floatingToolbar.confirmModalHeading',
    defaultMessage: 'Are you sure?',
    description: 'Default heading of floating toolbar confirmation modal.',
  },
  confirmModalOK: {
    id: 'fabric.editor.floatingToolbar.confirmModalOK',
    defaultMessage: 'OK',
    description: 'OK button for floating toolbar confirmation modal.',
  },
  confirmModalCancel: {
    id: 'fabric.editor.floatingToolbar.confirmModalCancel',
    defaultMessage: 'Cancel',
    description: 'Cancel button for floating toolbar confirmation modal.',
  },
  floatingToolbarAriaLabel: {
    id: 'fabric.editor.floatingToolbar.floatingToolbarAriaLabel',
    defaultMessage: 'Floating Toolbar',
    description: `a floating toolbar's aria label`,
  },
  floatingToolbarAnnouncer: {
    id: 'fabric.editor.floatingToolbar.floatingToolbarAnnouncer',
    defaultMessage: 'Floating toolbar controls have been opened',
    description: `message that will be announced to screenreaders that the floating toolbar is opened`,
  },
  floatingToolbarScrollLeft: {
    id: 'fabric.editor.floatingToolbar.scrollLeft',
    defaultMessage: 'Scroll left',
    description: `Button to scroll left when the toolbar is in the overflow state`,
  },
  floatingToolbarScrollRight: {
    id: 'fabric.editor.floatingToolbar.scrollRight',
    defaultMessage: 'Scroll right',
    description: `Button to scroll right when the toolbar is in the overflow state`,
  },
});
