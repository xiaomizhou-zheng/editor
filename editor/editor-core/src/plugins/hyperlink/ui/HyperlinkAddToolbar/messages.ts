import { defineMessages } from 'react-intl-next';

export const utilMessages = defineMessages({
  hyperlinkIconIssueLabel: {
    id: 'fabric.editor.headingLink.hyperlinkIconIssueLabel',
    defaultMessage: 'Issue',
    description: 'icon label for a jira issue',
  },
  hyperlinkIconBugLabel: {
    id: 'fabric.editor.headingLink.hyperlinkIconBugLabel',
    defaultMessage: 'Bug',
    description: 'icon label for a jira bug',
  },
  hyperlinkIconStoryLabel: {
    id: 'fabric.editor.headingLink.hyperlinkIconStoryLabel',
    defaultMessage: 'Story',
    description: 'icon label for a jira story',
  },
  hyperlinkIconTaskLabel: {
    id: 'fabric.editor.headingLink.hyperlinkIconTaskLabel',
    defaultMessage: 'Task',
    description: 'icon label for a jira task',
  },
  hyperlinkIconPageLabel: {
    id: 'fabric.editor.headingLink.hyperlinkIconPageLabel',
    defaultMessage: 'Page',
    description: 'icon label for a confluence page',
  },
  hyperlinkIconBlogLabel: {
    id: 'fabric.editor.headingLink.hyperlinkIconBlogLabel',
    defaultMessage: 'Blog',
    description: 'icon label for a confluence blog',
  },
});
