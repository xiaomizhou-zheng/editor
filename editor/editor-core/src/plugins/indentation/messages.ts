// Common Translations will live here
import { defineMessages } from 'react-intl-next';

export const messages = defineMessages({
  indent: {
    id: 'fabric.editor.indent',
    defaultMessage: 'Indent',
    description: 'Indent a list item, paragraph or heading',
  },
  outdent: {
    id: 'fabric.editor.outdent',
    defaultMessage: 'Outdent',
    description: 'Outdent a list item, paragraph or heading',
  },
});
