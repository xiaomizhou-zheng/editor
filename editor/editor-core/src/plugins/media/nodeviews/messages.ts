import { defineMessages } from 'react-intl-next';

export const messages = defineMessages({
  mediaGroupDeleteLabel: {
    id: 'fabric.editor.mediaGroupDeleteLabel',
    defaultMessage: 'delete',
    description: 'a label for an icon that deletes media',
  },
});
