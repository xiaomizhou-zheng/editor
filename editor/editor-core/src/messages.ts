// Common Translations will live here
import { defineMessages } from 'react-intl-next';
import commonMessages, { linkMessages } from '@atlaskit/editor-common/messages';
import { messages as statusMessages } from './plugins/status/nodeviews/messages';
import { messages as dateMessages } from './plugins/date/nodeviews/messages';

export { linkMessages };
export { statusMessages };
export { dateMessages };

export const linkToolbarMessages = defineMessages({
  addLink: {
    id: 'fabric.editor.mediaAddLink',
    defaultMessage: 'Add link',
    description: 'Add link',
  },
  unableToOpenLink: {
    id: 'fabric.editor.unableToOpenLink',
    defaultMessage: 'Unable to open this link',
    description: 'Unable to open this link',
  },
  unlink: {
    id: 'fabric.editor.unlink',
    defaultMessage: 'Unlink',
    description: 'Removes the hyperlink but keeps your text.',
  },
  editLink: {
    id: 'fabric.editor.editLink',
    defaultMessage: 'Edit link',
    description: 'Edit the link, update display text',
  },
  placeholder: {
    id: 'fabric.editor.hyperlinkToolbarPlaceholder',
    defaultMessage: 'Paste or search for link',
    description: 'Paste or search for link',
  },
  linkPlaceholder: {
    id: 'fabric.editor.linkPlaceholder',
    defaultMessage: 'Paste link',
    description: 'Create a new link by pasting a URL.',
  },
  linkAddress: {
    id: 'fabric.editor.linkAddress',
    defaultMessage: 'Link address',
    description: 'Insert the address of the link',
  },
  invalidLink: {
    id: 'fabric.editor.invalidLink',
    defaultMessage: 'Please enter a valid link.',
    description: 'Please enter a valid link.',
  },
  emptyLink: {
    id: 'fabric.editor.emptyLink',
    defaultMessage: 'Please enter a link.',
    description: 'Please enter a link.',
  },
  settingsLink: {
    id: 'fabric.editor.settingsLinks',
    defaultMessage: 'Go to Link Preferences',
    description: 'Go to Link Preferences',
  },
});

export default commonMessages;
