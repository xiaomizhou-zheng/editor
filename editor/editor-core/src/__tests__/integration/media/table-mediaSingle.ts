import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  editable,
  getDocFromElement,
  insertMedia,
} from '@atlaskit/editor-test-helpers/integration/helpers';
import {
  goToEditorTestingWDExample,
  mountEditor,
} from '@atlaskit/editor-test-helpers/testing-example-page';
import {
  Appearance,
  editorCommentContentSelector,
} from '@atlaskit/editor-test-helpers/vr-utils/base-utils';
import { messages as insertBlockMessages } from '../../../plugins/insert-block/ui/ToolbarInsertBlock/messages';

const editors: { appearance: Appearance; snapshotSelector?: string }[] = [
  { appearance: Appearance.fullPage },
  {
    appearance: Appearance.comment,
    snapshotSelector: editorCommentContentSelector,
  },
];

editors.forEach((editor) => {
  BrowserTestCase(
    `Media tables : ${editor.appearance} editor, can insert into table`,
    { skip: [] },
    async (client: any, testName: string) => {
      const page = await goToEditorTestingWDExample(client);
      await mountEditor(page, {
        appearance: editor.appearance,
        allowTables: {
          advanced: true,
        },
        media: {
          allowMediaSingle: true,
          allowMediaGroup: true,
        },
      });
      await page.click(
        `[aria-label="${insertBlockMessages.table.defaultMessage}"]`,
      );
      // second cell
      await page.keys('ArrowDown');

      await insertMedia(page);
      expect(await page.isVisible('[data-testid="media-file-card-view"]')).toBe(
        true,
      );

      const doc = await page.$eval(editable, getDocFromElement);
      expect(doc).toMatchCustomDocSnapshot(testName);
    },
  );
});
