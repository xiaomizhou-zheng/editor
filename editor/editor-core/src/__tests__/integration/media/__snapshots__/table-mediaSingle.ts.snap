// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media tables : comment editor, can insert into table 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
        "localId": "abc-123",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "layout": "align-start",
                    "width": null,
                  },
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "__contextId": "DUMMY-OBJECT-ID",
                        "__displayType": null,
                        "__external": false,
                        "__fileMimeType": "image/jpeg",
                        "__fileName": "example.png",
                        "__fileSize": 756325,
                        "alt": "",
                        "collection": "MediaServicesSample",
                        "height": 636,
                        "id": "1234-5678-abcd-efgh",
                        "occurrenceKey": null,
                        "type": "file",
                        "url": null,
                        "width": 640,
                      },
                      "type": "media",
                    },
                  ],
                  "type": "mediaSingle",
                },
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`Media tables : full-page editor, can insert into table 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
        "localId": "abc-123",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "layout": "center",
                    "width": null,
                  },
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "__contextId": "DUMMY-OBJECT-ID",
                        "__displayType": null,
                        "__external": false,
                        "__fileMimeType": "image/jpeg",
                        "__fileName": "example.png",
                        "__fileSize": 756325,
                        "alt": "",
                        "collection": "MediaServicesSample",
                        "height": 636,
                        "id": "1234-5678-abcd-efgh",
                        "occurrenceKey": null,
                        "type": "file",
                        "url": null,
                        "width": 640,
                      },
                      "type": "media",
                    },
                  ],
                  "type": "mediaSingle",
                },
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;
